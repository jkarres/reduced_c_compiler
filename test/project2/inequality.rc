// testing float/float, float/int, int/float inequalities

function : void main() {
  int i = 6, j = 7;
  float f = 6.0, g = 7.0;

  // unequal float and int
  // f=6.0, j=7

  cout << 1 << " " << f < f << endl;
  cout << 2 << " " << f < j << endl;
  cout << 3 << " " << j < f << endl;
  cout << 4 << " " << j < j << endl;

  cout << 5 << " " << f <= f << endl;
  cout << 6 << " " << f <= j << endl;
  cout << 7 << " " << j <= f << endl;
  cout << 8 << " " << j <= j << endl;

  cout << 9 << " " << f > f << endl;
  cout << 10 << " " << f > j << endl;
  cout << 11 << " " << j > f << endl;
  cout << 12 << " " << j > j << endl;

  cout << 13 << " " << f >= f << endl;
  cout << 14 << " " << f >= j << endl;
  cout << 15 << " " << j >= f << endl;
  cout << 16 << " " << j >= j << endl;

  // equal int and float
  // i=6, g=6.0

  cout << 17 << " " << i < i << endl;
  cout << 18 << " " << i < g << endl;
  cout << 19 << " " << g < i << endl;
  cout << 20 << " " << g < g << endl;

  cout << 21 << " " << i <= i << endl;
  cout << 22 << " " << i <= g << endl;
  cout << 23 << " " << g <= i << endl;
  cout << 24 << " " << g <= g << endl;

  cout << 25 << " " << i > i << endl;
  cout << 26 << " " << i > g << endl;
  cout << 27 << " " << g > i << endl;
  cout << 28 << " " << g > g << endl;

  cout << 29 << " " << i >= i << endl;
  cout << 30 << " " << i >= g << endl;
  cout << 31 << " " << g >= i << endl;
  cout << 32 << " " << g >= g << endl;

  // unequal floats
  // f=6.0, g=7.0

  cout << 33 << " " << f < f << endl;
  cout << 34 << " " << f < g << endl;
  cout << 35 << " " << g < f << endl;
  cout << 36 << " " << g < g << endl;

  cout << 37 << " " << f <= f << endl;
  cout << 38 << " " << f <= g << endl;
  cout << 39 << " " << g <= f << endl;
  cout << 40 << " " << g <= g << endl;

  cout << 41 << " " << f > f << endl;
  cout << 42 << " " << f > g << endl;
  cout << 43 << " " << g > f << endl;
  cout << 44 << " " << g > g << endl;

  cout << 45 << " " << f >= f << endl;
  cout << 46 << " " << f >= g << endl;
  cout << 47 << " " << g >= f << endl;
  cout << 48 << " " << g >= g << endl;

}
