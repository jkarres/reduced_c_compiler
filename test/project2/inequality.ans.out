1 false
2 true
3 false
4 false
5 true
6 true
7 false
8 true
9 false
10 false
11 true
12 false
13 true
14 false
15 true
16 true
17 false
18 true
19 false
20 false
21 true
22 true
23 false
24 true
25 false
26 false
27 true
28 false
29 true
30 false
31 true
32 true
33 false
34 true
35 false
36 false
37 true
38 true
39 false
40 true
41 false
42 false
43 true
44 false
45 true
46 false
47 true
48 true
